### PR TITLE
Feature/6747 health certificate service protocol and data models

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Bringing the protocol and data models so the coordinator can build on top of them. Errors are still in the works and will get a localized description that can be shown to the user.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6892

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/FetchProofCertificateTrigger.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+import HealthCertificateToolkit
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/FetchProofCertificateTrigger.swift:7]: Are those needed for an enum?

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateServiceError.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+import HealthCertificateToolkit
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateServiceError.swift:7]: Same here

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift:6]: Needed?

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/ProofCertificate.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/ProofCertificate.swift:6]: Needed?

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift:6]: `Never` comes with Foundation right?

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/Name+Extension.swift [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+			.filter { $0.trimmingCharacters(in: .whitespacesAndNewlines) != "" }
+			.joined(separator: " ")
+	}
+
+	var standardizedName: String {
+		[standardizedGivenName, standardizedFamilyName]
+			.compactMap { $0 }
+			.filter { $0.trimmingCharacters(in: .whitespacesAndNewlines) != "" }
+			.joined(separator: " ")
+	}
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/Name+Extension.swift:33]: Why not use a PersonNameComponentsFormatter() therefore?

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertifiedPerson.swift [hunk 7]
```diff
@@ -0,0 +1,47 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+class HealthCertifiedPerson: OpenCombine.ObservableObject, Codable, Equatable {
+
+	init(healthCertificates: [HealthCertificate], proofCertificate: ProofCertificate?) {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertifiedPerson.swift:10]:
```suggestion
        // MARK: - init

	init(healthCertificates: [HealthCertificate], proofCertificate: ProofCertificate?) {
```

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import HealthCertificateToolkit
+
+class MockHealthCertificateService: HealthCertificateServiceProviding {
+
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift:10]: MARKS please

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/FetchProofCertificateTrigger.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/FetchProofCertificateTrigger.swift]: Removed

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateServiceError.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateServiceError.swift]: Removed

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/ProofCertificate.swift (hunk 4 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/ProofCertificate.swift]: Removed

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift (hunk 5 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift:5]: Seems like its nod needed to import Foundation here :)

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/Name+Extension.swift (hunk 6 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/Name+Extension.swift:33]: I considered it, but the tech spec was very specific about the name formatting so I followed it. We could think about this again for the shown name, the standardized name is a more technical name so I think we definitely should follow the spec there. Leaving it as it is for now.

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertifiedPerson.swift (hunk 7 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertifiedPerson.swift:12]: Added

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift (hunk 8 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/MockHealthCertificateService.swift:9]: Added marks 👍

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/Name+Extension.swift (hunk 6 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/Name+Extension.swift:33]: 👍

**Pull request merged**